crates/usvg-parser/src/lib.rs

```diff
@@ -89,6 +136,46 @@ impl std::fmt::Display for Error {
             Error::ParsingFailed(ref e) => {
                 write!(f, "SVG data parsing failed cause {}", e)
             }
+            Error::UnexpectedEndOfStream => {
+                write!(f, "unexpected end of stream")
+            }
+            Error::UnexpectedData(pos) => {
+                write!(f, "unexpected data at position {}", pos)
+            }
+            Error::InvalidValue => {
+                write!(f, "invalid value")
+            }
+            Error::InvalidIdent => {
+                write!(f, "invalid ident")
+            }
+            Error::InvalidChar(ref chars, pos) => {
+                // Vec<u8> -> Vec<String>
+                let list: Vec<String> = chars
+                    .iter()
+                    .skip(1)
+                    .map(|c| String::from_utf8(vec![*c]).unwrap())
+                    .collect();
+
+                write!(
+                    f,
+                    "expected '{}' not '{}' at position {}",
+                    list.join("', '"),
+                    chars[0] as char,
+                    pos
+                )
+            }
+            Error::InvalidString(ref strings, pos) => {
+                write!(
+                    f,
+                    "expected '{}' not '{}' at position {}",
+                    strings[1..].join("', '"),
```

Comment:
Square brackets here concern me unless we have some guarantee that `strings` has a length of at least 1.